### PR TITLE
Add script to automatically update core packages

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
This PR adds a script that automatically updates core WordPress packages to their latest version.

Patch with updates generated using this script https://core.trac.wordpress.org/ticket/49142 (PR: https://github.com/WordPress/wordpress-develop/pull/128)

## How has this been tested?
I opened the WordPress develop dir on the terminal.
I executed "/.../gutenberg/bin/commander.js update-core" against that folder and verified the package.json package-lock.json and the block PHP files were changed.

### CODE REVIEW

File: bin/commander.js [hunk 1]
```diff
@@ -11,11 +11,16 @@ const chalk = require( 'chalk' );
...
-const childProcess = require( 'child_process' );
 const Octokit = require( '@octokit/rest' );
 const os = require( 'os' );
 const uuid = require( 'uuid/v4' );
 
+/**
+ * Internal dependencies
+ */
+const { readJSONFile, runShellScript, askForConfirmationToContinue } = require( './utils' );
+const { updateCorePackages } = require( './update-core-packages' );
```
Comment (youknowriad) [inline bin/commander.js]: Why this command is separate but not the others? Should we be consistent? 

Should we create a dedicated folder for this tool, now that it's split into multiple files?

**Review by youknowriad — COMMENTED** [review]
I didn't test but this is cool.

File: bin/commander.js (hunk 1 above)
Comment (jorgefilipecosta) [reply bin/commander.js]: > Why this command is separate but not the others? Should we be consistent?

The main file got too big, I thought it was not a good idea to add the implementation of another command there. I think we should reorganize and split the commands into multiple files, I avoided doing this here to make reviewing easier. As soon as we land this PR I plan on proposing a PR that refactors our commands.

**Comment by gziolo** [comment]
I have just tested the WordPress core patch and everything works perfectly fine. Excellent work on this tool. As discussed in the private chat, it feels like it would be easier to have a way to run this script directly from `wordpress-develop` repository, knowing that you still need to run `npm run build` and sometimes apply some manual changes afterwards.

**Comment by youknowriad** [comment]
I think this release tool entirely could be a separate repo/tool/package instead of just a script in the gutenberg repo.

**Comment by youknowriad** [comment]
I feel we could just update the script a bit to generate a `.diff` file and give you the patch so you can apply it wherever you want.

**Comment by gziolo** [comment]
> I feel we could just update the script a bit to generate a `.diff` file and give you the patch so you can apply it wherever you want.

I love the idea, this way we could use it with all projects that use WordPress packages. It could also become a way to self-update `@wordpress/scripts` by running `wp-scripts packages-update` :)

**Comment by jorgefilipecosta** [comment]
Hi @youknowriad, @gziolo,

I have considered the idea of generating a .diff file but it is much slower because one needs to clone the wordpress-develop repository again (to not depend on the user having it previously cloned) and given that to generate the diff we need npm install and npm run build we would always need to do a full npm install on a blank repository which is very slow (e.g: chromium is downloaded).


@gziolo I like the idea of making this part of @wordpress/scripts so other plugins/third parties can use this to update their package versions.
I will look into how I could make this part of wp-scripts and use that in core.

**Comment by gziolo** [comment]
> I will look into how I could make this part of wp-scripts and use that in core.

I'm looking forward to it, thank you for working on this script. It's going to be totally awesome :)

**Comment by jorgefilipecosta** [comment]
This PR was updated to follow @gziolo suggestion, and the script is now part of @wordpress/scripts.

I tested the script by setting the current path on the terminal to the WordPress develop directory and then executing "/Users/.../gutenberg/packages/scripts/bin/wp-scripts.js packages-update" followed by "npm run build". The generated patch can be checked in https://core.trac.wordpress.org/ticket/49204 and https://github.com/WordPress/wordpress-develop/pull/132.

The plan is to make the script easier to use. After the next package release we should add the following config in core:
```
{
  "scripts": {
    "packages-update": "wp-scripts packages-update",
    "postpackages-update": "npm run build"
  }
}
```
So we can simply do npm run packages-update in core to update the WordPress packages.


Third-party users can easily do the same by importing @wordpress/scripts.

**Review by gziolo — APPROVED** [review]
Awesome work. It still needs to be documented in the README file and includes a new feature in the CHANGELOG file. You pick how you proceed, one or two PRs 🚢

**Comment by youknowriad** [comment]
It would be good to rewrite some blog posts about the scripts package

**Comment by gziolo** [comment]
> It would be good to rewrite some blog posts about the scripts package

@youknowriad – can you expand what should be covered? I'd be more than happy to take care of it.

**Comment by youknowriad** [comment]
We could write a series of posts:

 - How the setup, 
 - how to use the packages and update them
 - how to build
 - how to test
 - how to lint and format
  ...

I know we have some of these in docs, but the blog post format works well I think for awareness.

**Pull request merged**